lib/libc/net/lib_inetaddr.c

```diff
@@ -92,7 +92,30 @@ in_addr_t inet_addr(FAR const char *cp)
 
 int inet_aton(const char *cp, struct in_addr *inp)
 {
+	int idx = 0;
+	int dot_count = 0;
+	int check_range[4] = { -1, -1, -1, -1 };
+	while (cp[idx] != '\0') {
+		if ((cp[idx] < '0' || cp[idx] > '9') && cp[idx] != '.') {
+			return 0;
+		}
+		if (idx > 14) {
```

Comment:
when there are some if - return states, you'd better check cheapest one first.
in this case
if (idx > 14) {
     return 0;
}

below also same meaning.
if ((cp[idx] != '.' && ( cp[idx] < '0' || cp[idx] > '0')) {
    return 0;
}

It is not big impact here but this is basic habit to improve performance.